### PR TITLE
Update golang to 1.19.2

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
This PR update the go compiler from version 1.17 to the latest 1.19 release.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Using `io/ioutil` now breaks the linter, so I replaced all uses of it with functions from `io`/`os`

**Release note**:

```release-note
Built with golang 1.19
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -207,4 +207,4 @@ replace (
 	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.6.2
 )
 
-go 1.17
+go 1.19
```
Comment (rmohr) [inline go.mod:210]: Bazel will still use 1.17. You have to update the WORKSPACE dependencies (rules_go and gazelle) and bump the golang version: https://github.com/bazelbuild/bazel-gazelle#running-gazelle-with-bazel.

**Comment by rmohr** [comment]
> Using io/ioutil now breaks the linter, so I replaced all uses of it with functions from io/os

Could you share the issue it produces? Maybe that is because bazel was not updated  to 1.19?

**Comment by rmohr** [comment]
/hold

bazel is not bumpte to 1.19

File: go.mod (hunk 1 above)
Comment (jean-edouard) [reply go.mod:210]: Ah, thank you, I knew it couldn't be that easy, working on it now!

**Comment by jean-edouard** [comment]
> > Using io/ioutil now breaks the linter, so I replaced all uses of it with functions from io/os
> 
> Could you share the issue it produces? Maybe that is because bazel was not updated to 1.19?

Here: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8622/pull-kubevirt-code-lint/1582365920414666752

**Comment by iholder-redhat** [comment]
/cc
p.s. yay, generics! :smile:

**Comment by rmohr** [comment]
@jean-edouard pushed the necessary bazel changes to your branch. I think we need to bump the builder image once more.

**Comment by jean-edouard** [comment]
> @jean-edouard pushed the necessary bazel changes to your branch. I think we need to bump the builder image once more.

Wow, thank you so much! I'll take care of the bump asap today

File: WORKSPACE [hunk 2]
```diff
@@ -4,6 +4,8 @@ load("//third_party:deps.bzl", "deps")
 
 deps()
 
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
```
Comment (rmohr) [inline WORKSPACE:7]: ok, that one is not needed. An oversight on my part. It is imported below already.

**Comment by rmohr** [comment]
On the code-coverage job I see

```
2022/10/24 13:48:52 failed to parse profiles: bad mode line: SF:staging/src/kubevirt.io/client-go/log/log.go
```

which may indicate that our coverage collector needs a fix too.

Retesting  to confirm.

/test pull-kubevirt-goveralls

**Comment by jean-edouard** [comment]
> On the code-coverage job I see
> 
> ```
> 2022/10/24 13:48:52 failed to parse profiles: bad mode line: SF:staging/src/kubevirt.io/client-go/log/log.go
> ```
> 
> which may indicate that our coverage collector needs a fix too.
> 
> Retesting to confirm.
> 
> /test pull-kubevirt-goveralls

Indeed...
We use this tool, last updated 7 years ago! https://github.com/wadey/gocovmerge
Not sure if there's some new tool we could switch to for this.

**Comment by rmohr** [comment]
> Indeed...
> We use this tool, last updated 7 years ago! https://github.com/wadey/gocovmerge
> Not sure if there's some new tool we could switch to for this.
> Edit: Bazel seems to have a (new?) option for this: https://bazel.build/configure/coverage#running_coverage

We use bazel to create the coverage files. But bazel creates coverage files for each individual target and does not merge it (well maybe in 5.x it does, need to check). Therefore we use gocovmerge. There is probably just something in a generated file it does not like. Should be easy to fix. Will check asap.

**Comment by jean-edouard** [comment]
> > Indeed...
> > We use this tool, last updated 7 years ago! https://github.com/wadey/gocovmerge
> > Not sure if there's some new tool we could switch to for this.
> > Edit: Bazel seems to have a (new?) option for this: https://bazel.build/configure/coverage#running_coverage
> 
> We use bazel to create the coverage files. But bazel creates coverage files for each individual target and does not merge it (well maybe in 5.x it does, need to check). Therefore we use gocovmerge. There is probably just something in a generated file it does not like. Should be easy to fix. Will check asap.

I tried that bazel option for unified output still got the error, so it must be coming from the goveralls tool itself. Trying to update it now.

**Comment by jean-edouard** [comment]
> /hold
> 
> bazel is not bumpte to 1.19

Everything should be bumped now, thanks @rmohr !
/hold cancel

**Comment by rmohr** [comment]
/override pull-kubevirt-e2e-k8s-1.24-sig-storage-dv-gc

The test got removed.

**Pull request merged**